components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/UtxoSignedTransactionImpl.kt

```diff
@@ -112,34 +113,57 @@ data class UtxoSignedTransactionImpl(
         )
     }
 
+    private fun getSignatoryKeyFromKeyId(keyId: SecureHash): PublicKey? {
+        val keyIdsToSignatories = signatories.associateBy {// todo cache this
+            transactionSignatureServiceInternal.getIdOfPublicKey(
+                it,
+                keyId.algorithm
+            )
+        }
+        return keyIdsToSignatories[keyId]
+    }
+
+    // Against signatories. Notary/Unknown signatures are ignored.
+    @Suspendable //TODO are these need to be suspendable?
     override fun getMissingSignatories(): Set<PublicKey> {
-        val appliedSignatories = signatures.filter {
-            try {
-                transactionSignatureService.verifySignature(this, it)
-                true
-            } catch (e: Exception) {
-                false
+        val appliedSignatures = signatures.mapNotNull {
+            val publicKey = getSignatoryKeyFromKeyId(it.by)
+            if (publicKey == null) {
+                null
+            } else {
+                try {
+                    transactionSignatureServiceInternal.verifySignature(this, it, publicKey)
+                    publicKey
+                } catch (e: Exception) {
+                    null
+                }
             }
-        }.map { it.by }.toSet()
+        }.toSet()
 
         // isFulfilledBy() helps to make this working with CompositeKeys.
-        return signatories.filterNot { KeyUtils.isKeyFulfilledBy(it, appliedSignatories) }.toSet()
+        return signatories.filterNot { KeyUtils.isKeyFulfilledBy(it, appliedSignatures) }.toSet()
     }
 
+    // Against signatories. Notary/unknown signatures are ignored
     @Suspendable
-    override fun verifySignatures() {
-        val appliedSignatories = signatures.filter {
-            try {
-                transactionSignatureService.verifySignature(this, it)
-                true
-            } catch (e: Exception) {
-                throw TransactionSignatureException(
-                    id,
-                    "Failed to verify signature of ${it.signature} for transaction $id. Message: ${e.message}",
-                    e
-                )
+    override fun verifySignatorySignatures() {
+        val appliedSignatories = signatures.mapNotNull {
+            val publicKey = getSignatoryKeyFromKeyId(it.by)
+            if (publicKey == null) {
+                null// We do not care about non-notary/non-signatory keys
+            } else {
+                try {
+                    transactionSignatureServiceInternal.verifySignature(this, it, publicKey)
+                    publicKey
+                } catch (e: Exception) {
+                    throw TransactionSignatureException(
+                        id,
+                        "Failed to verify signature of ${it.signature} for transaction $id. Message: ${e.message}",
```

Comment:
I think this should cover the negative cases (re logging):
```suggestion
                        "Failed to verify signature of ${it.signature} from $publicKey for transaction $id. Message: ${e.message}",
```